CodelabsOne/app/src/main/java/com/feylabs/codelab_one/View/ListViewActivity.kt

```diff
@@ -0,0 +1,55 @@
+package com.feylabs.codelab_one.View
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Toast
+import com.feylabs.codelab_one.Adapter.BookListAdapter
+import com.feylabs.codelab_one.Data.BookDataSource
+import com.feylabs.codelab_one.Model.BookModel
+import com.feylabs.codelab_one.ViewModel.BookViewModel
+import com.feylabs.codelab_one.databinding.ActivityListViewBinding
+
+class ListViewActivity : AppCompatActivity() {
+    lateinit var viewBInding: ActivityListViewBinding
+    lateinit var bookAdapter: BookListAdapter
+    lateinit var bookViewModel: BookViewModel
+
+    var tempBookData = mutableListOf<BookModel>()
+
+    companion object{
+        const val  BOOK_PARCEL = "JNE DISINI"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewBInding = ActivityListViewBinding.inflate(layoutInflater)
+        setContentView(viewBInding.root)
+
+        tempBookData.addAll(BookDataSource.bookData)
```

Comment:
better use add All than 
tempBookData = (BookDataSource.bookData)
i will try the difference